### DISCUSSION TITLE
More than 500 results?

### DISCUSSION THREAD

**Discussion by openaudible** [mkb79/Audible#119] [General]
It looks like the audible api/html requests are limited to about 500 results. Any page requested after that number returns the last batch of results. (Request for page >10 returns page 10 results).

For example: 
```
https://api.audible.com/1.0/catalog/products?response_groups=product_desc%2C&page=1&num_results=50
https://api.audible.com/1.0/catalog/products?response_groups=product_desc%2C&page=9&num_results=50
https://api.audible.com/1.0/catalog/products?response_groups=product_desc%2C&page=10&num_results=50
```
All work fine.
```
https://api.audible.com/1.0/catalog/products?response_groups=product_desc%2C&page=11&num_results=50
```
Returns the same results as page 10.

Is there a good way to get more than 500 results?
Or get a list of all of the ASIN's?

**Comment by mkb79** [comment]
That is well known. It is the normal behavior of the API. There is a way to request (nearly) all products in the API. But this way is not open source. Sorry.

**Reply by openaudible** [reply]
That's what I figured, and definitely curious.  A (public) BigQuery database would be nice!

I tried catalog/products with products_since_timestamp=2021-03-20T18:01:47.935Z (escaped) and other random dates and it doesn't seem to change the query results. Maybe purposefully broken for the same reason?

Thanks for your work! Unfortunately I don't know python very well, so not much of a contributor.

**Reply by mkb79** [reply]
The problem with the Audible API is the lack of documentation. That's why sometimes you have to get creative to get the desired result. It took me some time to find a method to get almost all the data from the catalog/products endpoint. I‘m use these method only for my private use. I does not want that everyone request the API to rebuild the entire Audible catalog and Audible/Amazon are forced to change the API or take other security measures.

**Reply by Meivyn** [reply]
Is there any chance you could share this method privately? I'd especially like to use it to search through specific keywords or things like that when I'm looking for a specific type of book. Audible search is awful to say the least. I can still try to reverse myself but figured it doesn't hurt to ask.
